crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -581,9 +583,31 @@ impl Format<PyFormatContext<'_>> for DocstringStmt<'_> {
                     constant
                         .format()
                         .with_options(ExprConstantLayout::String(StringLayout::DocString)),
-                    trailing_comments(node_comments.trailing),
                 ]
-            )
+            )?;
+
+            // Comments after docstrings need a newline between the docstring and the comment, so we attach
+            // it as leading on the first statement after the docstring.
+            // (https://github.com/astral-sh/ruff/issues/7948)
+            // ```python
+            // class ModuleBrowser:
+            //     """Browse module classes and functions in IDLE."""
+            //     # ^ Insert a newline above here
+            //
+            //     def __init__(self, master, path, *, _htest=False, _utest=False):
+            //         pass
+            // ```
+            if let Some(own_line) = node_comments
+                .trailing
+                .iter()
+                .find(|comment| comment.line_position().is_own_line())
+            {
+                if lines_before(own_line.start(), f.context().source()) < 2 {
+                    empty_line().fmt(f)?;
+                }
+            }
+
+            trailing_comments(node_comments.trailing).fmt(f)
```

Comment:
This should only be applied to class docstrings, right? (Wouldn't it also apply to function docstrings as written?)

Reply:
@konstin - Sorry to ping, did this get addressed? I don't see test cases for it.

Reply:
Addressed in https://github.com/astral-sh/ruff/pull/8375